x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -291,7 +291,8 @@ fragment ENRICH_POLICY_NAME_BODY
     ;
 
 ENRICH_POLICY_NAME
-    : (LETTER | DIGIT) ENRICH_POLICY_NAME_BODY*
+    // allow prefix for the policy to specify its resolution
+    : (ENRICH_POLICY_NAME_BODY+ COLON)? ENRICH_POLICY_NAME_BODY+
```

Comment:
I guess changing the allowed characters here is on purpose.
Before this change, the policy name only allowed letters or numbers as first char, now any character apart from those explicitly excluded in `ENRICH_POLICY_NAME_BODY` is allowed.
The good is that it allows all the valid policy names to be used (eg. a policy name can start with `$` or `.`)
The bad is that it also allows invalid policy names, eg. `.`, `..` or names starting with `+`, `_`, `-`

The impact is limited though, because the validation takes care of checking that the policy does not exist.
Most of the problem is for Kibana, that will need further validation outside of the grammar